File: app/src/main/java/de/szalkowski/activitylauncher/IconListAsyncProvider.java
package de.szalkowski.activitylauncher;

import android.content.Context;

public class IconListAsyncProvider extends AsyncProvider<IconListAdapter> {
	public IconListAsyncProvider(Context context, Listener<IconListAdapter> listener) {
		super(context, listener, false);
	}
	
	@Override
	protected IconListAdapter run(Updater updater) {
		IconListAdapter adapter = new IconListAdapter(this.context, updater);
		return adapter;
	}
}

File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/IconListAsyncProvider.kt
package de.szalkowski.activitylauncher

import android.content.Context

internal class IconListAsyncProvider(context: Context, listener: AsyncProvider.Listener<IconListAdapter>) : AsyncProvider<IconListAdapter>(context, listener, false) {
    private val adapter: IconListAdapter

    init {
        this.adapter = IconListAdapter(context)
    }

    override fun run(updater: AsyncProvider<IconListAdapter>.Updater): IconListAdapter {
        adapter.resolve(updater)
        return this.adapter
    }
}
